System/Vector.h

```diff
@@ -543,6 +545,10 @@ namespace RTE {
 		/// Clears all the member variables of this Vector, effectively resetting the members of this abstraction level only.
 		/// </summary>
 		void Clear() { m_X = m_Y = 0; }
+
+		// TODO: Investigate, based on where this value is used, whether making it Larger (by one or two orders of magnitude) can produce more desirable results.
+		float m_FloatTolerance = 0.0000001F; //!< Floating point comparison tolerance.
```

Comment:
Might be worth expanding the member var comment on this to explain what it's used for a little
i.e. `//!< Floating point comparison tolerance, for determining if a Vector's X or Y values should be considered to be zero.`

Might be overkill though, I'm okay with leaving it as-is if you'd rather not change it.

Reply:
that's a little too specific.
and now i'm realizing i missed using it in the operator overloading.
the general idea is that this tolerance would also be used when comparing vector components to other vector components. so for '==' operator instead of:
lhs.m_X == rhs.m_X
i would write:
fabs(lhs.m_X - rhs.m_X) < m_FloatTolerance;

Reply:
Ahh gotcha, that makes sense. Yeah, wasn't thinking there but of course it'd have to apply for all comparison operators. Maybe reword it a little then to describe what it does more, i.e. `Used to compare Float values in Vectors to determine equality.` or something?
Maybe it's fine as is though and I'm being silly.

Also yeah, please do include it with operator overloading.

Reply:
I've been reading a bit more about floating point number comparisons and I'd like to give it a better treatment in a separate pr.
The current way I handle tolerance, now I feel, isn't good enough and may give bad results.
Until it's done properly, I think comparisons should stay relative to absolute zero without tolerance, as was done until now.
Some references if you wanna see what I'm talking about:
[FP comparisons](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/)
[In short](http://realtimecollisiondetection.net/blog/?p=89)

Reply:
Interesting stuff, I've read the short one and I'll read the longer one over time. It does seem to me that in an either-or case between absolute and relative, absolute would be more helpful. While it doesn't do well for large scale numbers, the one thing we can be sure of is that number scale in CC is not very large and if two floats are not within a quite low absolute tolerance of each other, they shouldn't be considered equal.
Are you thinking of going for the two different tolerances solution?

Anyway, I agree that it's reasonable to split this work out into a separate PR and leave it untouched. Would you mind undoing your changes here?
Also, when you're redoing this I think it would be good to make and use some general-purpose float comparison function in RTETools (that's probably the most appropriate spot), since all this stuff can apply to floats that aren't unit components of Vectors.

Reply:
Haven't thought yet about how exactly I should do tolerance, I just saw that there's a good chance that doing only absolute tolerance isn't good enough. That was enough for me to decide it warrants more reading, to be done later.

Changes related to tolerance undone.

RTETools is exactly what i had in mind.

Reply:
Okay great, pop in a technical improvement issue to fix float tolerance and then feel free to start working on it. Maxim is trying to get his next PR done before he looks through this, but depending on how long that takes I might try and get him to get this done first.

Reply:
Done.
Issue #148.